app/controllers/incomeSources/add/IncomeSourceReportingMethodController.scala

```diff
@@ -115,7 +115,7 @@ class IncomeSourceReportingMethodController @Inject()(val authActions: AuthActio
           errorHandler(isAgent).showInternalServerError()
         }
       case true =>
-        itsaStatusService.hasMandatedOrVoluntaryStatusCurrentYear.flatMap {
+        itsaStatusService.hasMandatedOrVoluntaryStatusCurrentYear(user.nino).flatMap {
```

Comment:
these changes need to be confirm with someone from Solar team, as we not own this part of sub-domain